### PR TITLE
[14.0]  shopfloor_reception: prevent creating lines with negative demand

### PR DESCRIPTION
**Pull request by JuMiSanAr**
ref: rau-196

### CODE REVIEW

File: shopfloor/models/stock_move_line.py [hunk 1]
```diff
@@ -153,6 +153,7 @@ def _split_qty_to_be_done(self, qty_done, split_partial=True, **split_default_va
         :param split_partial: split if qty is less than expected
             otherwise rely on a backorder.
         """
+        assert self.product_uom_qty >= 0, "The demand cannot be negative"
```
Comment (simahawk) [inline shopfloor/models/stock_move_line.py]: you should raise a UserError

**Review by simahawk — CHANGES_REQUESTED** [review]

**Review by simahawk — APPROVED** [review]
LG

**Review by jbaudoux — APPROVED** [review]
LG

**Pull request merged**